src/components/WeatherComponent.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import axios, { AxiosResponse } from 'axios';
+import { weatherAPIKey } from '../key.json';
+
+const WeatherComponentAPI: React.FC<unknown> = () => {
+  const [city, setCity] = React.useState<string>('London');
+  const [search, setSearch] = React.useState<string>('London');
+  const [weather, setWeather] = React.useState<string>('');
+  const [weatherIcon, setWeatherIcon] = React.useState<string>('');
+  const [temperature, setTemperature] = React.useState<string>('');
+
+  React.useEffect(() => {
+    axios
+      .get<any, AxiosResponse<any, any>, any>(
+        `http://api.weatherapi.com/v1/current.json?key=${weatherAPIKey}&q=${city}`
+      )
+      .then(({ data }) => {
+        setTemperature(data.current.temp_c);
+        setWeather(data.current.condition.text);
+        setWeatherIcon(data.current.condition.icon);
+      });
+  }, [city]);
+
+  return (
+    <>
+      <h3>Weather API</h3>
+      <div>
+        <input
+          onChange={(event) => setSearch(event.target.value)}
```

Comment:
I think we could just ask the location info on browser rather than make them search every time..?
navigator.geolocation.getCurrentPosition(handleGeoSuccess, handleGeoError);